### ISSUE TITLE
Android crash when using 0.7.1 with React Native 0.32

### ISSUE THREAD

**Issue by chrisknepper** [lelandrichardson/react-native-maps#497]
Upon attempting to load any instance of a `MapView`, the map component is visible (but not filled-in) for a few seconds, but then the app crashes with the following stack trace:

```
08-25 11:54:58.281 893-893/com.lmt E/CMTUncaughtExceptionHandler: Thread name = main ex.message = No direct method <init>(IJ)V in class Lcom/facebook/react/uimanager/events/Event; or its super classes (declaration of 'com.facebook.react.uimanager.events.Event' appears in /data/app/com.lmt-2/base.apk) stack =java.lang.NoSuchMethodError: No direct method <init>(IJ)V in class Lcom/facebook/react/uimanager/events/Event; or its super classes (declaration of 'com.facebook.react.uimanager.events.Event' appears in /data/app/com.lmt-2/base.apk)
                                                                      at com.airbnb.android.react.maps.RegionChangeEvent.<init>(RegionChangeEvent.java:16)
                                                                      at com.airbnb.android.react.maps.AirMapView$8.onCameraChange(AirMapView.java:220)
                                                                      at com.google.android.gms.maps.GoogleMap$12.onCameraChange(Unknown Source)
                                                                      at com.google.android.gms.maps.internal.zze$zza.onTransact(Unknown Source)
                                                                      at android.os.Binder.transact(Binder.java:499)
                                                                      at wg.a(:com.google.android.gms.DynamiteModulesB:93)
                                                                      at maps.D.a.a(Unknown Source)
                                                                      at maps.D.a$1.run(Unknown Source)
                                                                  	at android.os.Handler.handleCallba
```

**Comment by alexHlebnikov** [comment]
With version of Google Play Services do you use? Try with latest one.

**Comment by chrisknepper** [comment]
@alexHlebnikov That doesn't work because `react-native-maps` lists Google Play Services 8.4.0 as a dependency in [`lib/build.gradle`](https://github.com/lelandrichardson/react-native-maps/blob/master/android/lib/build.gradle), so if I specify a newer version in my own `build.gradle`, compilation fails with:

```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:transformClassesWithJarMergingForDebug'.
> com.android.build.api.transform.TransformException: java.util.zip.ZipException: duplicate entry: com/google/android/gms/internal/zzpz.class
```

**Comment by alexHlebnikov** [comment]
Well, I'm using 9.4.0 and manually change it in component build.gradle and it works great. 
Also, check what you have installed and updated all Google SDK's for various Android versions and all the Tools.

**Comment by chrisknepper** [comment]
Can you post your `build.gradle` or tell me if you've made any changes to the way you're including `react-native-maps`? And are you truly using `react-native` version 0.32?

**Comment by alexHlebnikov** [comment]
Here is my app/build.gradle:
```
dependencies {
    compile project(':react-native-maps')
    compile fileTree(dir: 'libs', include: ['*.jar'])
    // From node_modules
    compile 'com.android.support:appcompat-v7:24.2.0'
    compile 'com.facebook.react:react-native:+'
    compile 'com.google.maps.android:android-maps-utils:0.4.3'
    compile 'com.google.android.gms:play-services-base:9.4.0'
    compile 'com.google.android.gms:play-services-maps:9.4.0'
}

...

apply plugin: 'com.google.gms.google-services'
```

And here is my android/build.gradle:
```
buildscript {
    repositories {
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:2.1.3'
        classpath 'com.google.gms:google-services:3.0.0'

        // NOTE: Do not place your application dependencies here; they belong
        // in the individual module build.gradle files
    }
}

allprojects {
    repositories {
        mavenLocal()
        jcenter()
        maven {
            // All of React Native (JS, Obj-C sources, Android binaries) is installed from npm
            url "$rootDir/../node_modules/react-native/android"
        }
    }
}
```

**Comment by chrisknepper** [comment]
Hmmm, I'm not sure why mine wasn't working in that case. It may be the line:

```groovy
apply plugin: 'com.google.gms.google-services'
```

But in either case, I just reverted back to Play Services 8.4 since I was having trouble getting the newer version of Play Services to work on my older Android phones. Along the way I discovered this snippet which did allow me to force 9.4 successfully:

```groovy
configurations.all {
  resolutionStrategy {
    force "com.google.android.gms:play-services-base:9.4.0"
    force "com.google.android.gms:play-services-maps:9.4.0"
    force "com.google.android.gms:play-services-analytics:9.4.0"
  }
}
```

**Comment by waqas19921** [comment]
Where to put that block?

**Comment by waqas19921** [comment]
I am getting this error after upgrade to react-native 0.32.0 in RegionChangeEvent.java
`Error:(16, 9) error: no suitable constructor found for Event(int,long)
constructor Event.Event() is not applicable
(actual and formal argument lists differ in length)
constructor Event.Event(int) is not applicable
(actual and formal argument lists differ in length)`

**Comment by rgurzhiy** [comment]
Hey guys.
The problem has nothing to do with Play Services.
Error message tells you, that compiler is trying to call `com.facebook.react.uimanager.events.Event`'s constructor `Event(int,long)` which is not present in RN 0.32.0 anymore (it has `Event(int)` constructor instead)
#490 - the same issue

**Comment by waqas19921** [comment]
Yes @rgurzhiy is right the problem is in only one file. that is in that pull request #450

**Comment by rgurzhiy** [comment]
Well. Actually, the problem is much more complex. 
There are at least 3-4 different issues with RN version compatibility:
1. Android: No direct method <init>(IJ)V in class Lcom/facebook/react/uimanager/events/Event
2. iOS: CTImageLoader interface no longer has loadImageWithTag.
3. JS: Warning: ReactClass: You're attempting to include a mixin that is either null or not an object.
4. Android: requirement of old Google Play Services version 8.4.0 (actual 9.4.0)
... It could be more
Unfortunately, your pull request is the solution for only one of them.
At the moment my fork solves this 4, but still it only can be considered as a temporary workaround.
Waiting for upgrade ))

**Comment by alexHlebnikov** [comment]
@rgurzhiy
Btw, as far as I know `NativeMethodsMixin` must exist in `react-native-maps` components files (you can read about it [here] (https://facebook.github.io/react-native/docs/nativemethodsmixin.html)). I can
t find it in your fork. 
You just need to add correct path to it: 
```
var NativeMethodsMixin = require('react/lib/NativeMethodsMixin');
```

**Comment by rgurzhiy** [comment]
@alexHlebnikov 
Thks for pointing that out. I'll try to fix it.

**Comment by chrisknepper** [comment]
@rgurzhiy @alexHlebnikov I solved the 0.32/Android problems in my pull request, check it out: #502 

If you change your `package.json` to what I have in that issue and `npm install`, both the Android and iOS problems you describe will be fixed because it will pull the project from Github instead of NPM (the iOS problem is already fixed in the master branch).

Hopefully the repo owners will merge the PR and then do a new release to NPM.

**Comment by spikebrehm** [comment]
Fixed in #502.
